runtime/client_http_request_test.go

```diff
@@ -62,6 +62,7 @@ func TestMakingClientWriteJSONWithBadJSON(t *testing.T) {
 		"clientID",
 		[]string{"DoStuff"},
 		"/",
+		map[string]string{},
```

Comment:
can we have test when header is not empty and verify it's being correctly passed in and set?

Reply:
just fyi, i added the test line #134
this test and the next are testing bad input so the request is never sent to the server.